plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergPageSourceProvider.java

```diff
@@ -269,11 +285,7 @@ private static ConnectorPageSource createOrcPageSource(
             OrcReader reader = OrcReader.createOrcReader(orcDataSource, options)
                     .orElseThrow(() -> new TrinoException(ICEBERG_BAD_DATA, "ORC file is zero length"));
             List<OrcColumn> fileColumns = reader.getRootColumn().getNestedColumns();
-            Map<Integer, OrcColumn> fileColumnsByIcebergId = fileColumns.stream()
-                    .filter(orcColumn -> orcColumn.getAttributes().containsKey(ORC_ICEBERG_ID_KEY))
-                    .collect(toImmutableMap(
-                            orcColumn -> Integer.valueOf(orcColumn.getAttributes().get(ORC_ICEBERG_ID_KEY)),
-                            identity()));
+            Map<Integer, OrcColumn> fileColumnsByIcebergId = mapIdsToOrcFileColumns(fileColumns);
             Map<String, OrcColumn> fileColumnsByName = null;
             if (fileColumnsByIcebergId.isEmpty()) {
```

Comment:
this is preexistent; but check for `empty` is not obvious to me. Should we rather check if we are missing mapping for some columns?

Reply:
I think the assumption is that if the Id field exists for one column it must exist for all of them. I haven't seen a case where they don't exist yet.

Reply:
Can we add a checkstate that it holds? So the result map is either empty or the size matches fileColumns?

Reply:
Well, columns can be missing if they've been added since this data file was created.

Reply:
> if an old table is migrated to iceberg without rewriting data (which is going to be most common scenario initially), files won't contain ids at all.

Are there any tests for that situation? Would like to add a projected test case

Reply:
I added some migrated cases to the Spark compatibility tests. @phd3 are you familiar with the Iceberg `schema.name-mapping.default` property? Looks like a more robust way we could be mapping column names for migrated tables.

Reply:
If we had the full Iceberg schema here as suggested in https://github.com/trinodb/trino/pull/8754 I think we could get the mapping from `schema.findField(String name)`